symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/datafeed/impl/AbstractDatafeedService.java

```diff
@@ -26,93 +32,91 @@
+    this.datafeedApi = datafeedApi;
+    this.listeners = new ArrayList<>();
+    this.authSession = authSession;
+    this.bdkConfig = config;
+  }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public void unsubscribe(RealTimeEventListener listener) {
-        listeners.remove(listener);
-    }
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void subscribe(RealTimeEventListener listener) {
+    listeners.add(listener);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void unsubscribe(RealTimeEventListener listener) {
+    listeners.remove(listener);
+  }
 
-    /**
-     * Handle a received listener by using the subscribed {@link RealTimeEventListener}.
-     *
-     * @param events List of Datafeed events to be handled
-     *
-     */
-    protected void handleV4EventList(List<V4Event> events) {
-        for (V4Event event : events) {
-            if (event == null || event.getType() == null) {
-                continue;
-            }
-            if (this.isSelfGeneratedEvent(event)) {
-                continue;
-            }
-            try {
-                RealTimeEventType eventType = RealTimeEventType.valueOf(event.getType());
-                for (RealTimeEventListener listener : listeners) {
-                    eventType.dispatch(listener, event);
-                }
-            } catch (IllegalArgumentException e) {
-                log.warn("Receive events with unknown type: {}", event.getType());
-            }
+  /**
+   * Handle a received listener by using the subscribed {@link RealTimeEventListener}.
+   *
+   * @param events List of Datafeed events to be handled
+   */
+  protected void handleV4EventList(List<V4Event> events) {
+    for (V4Event event : events) {
+      if (event == null || event.getType() == null) {
+        continue;
+      }
+      if (this.isSelfGeneratedEvent(event)) {
+        continue;
+      }
+      try {
+        RealTimeEventType eventType = RealTimeEventType.valueOf(event.getType());
+        for (RealTimeEventListener listener : listeners) {
+          eventType.dispatch(listener, event);
         }
+      } catch (IllegalArgumentException e) {
+        log.warn("Receive events with unknown type: {}", event.getType());
+      }
     }
+  }
 
-    private boolean isSelfGeneratedEvent(V4Event event) {
-        return event.getInitiator() != null && event.getInitiator().getUser() != null
-                && event.getInitiator().getUser().getUsername() != null
-                && event.getInitiator().getUser().getUsername().equals(this.bdkConfig.getBot().getUsername());
-    }
+  private boolean isSelfGeneratedEvent(V4Event event) {
+    return event.getInitiator() != null && event.getInitiator().getUser() != null
+        && event.getInitiator().getUser().getUsername() != null
+        && event.getInitiator().getUser().getUsername().equals(this.bdkConfig.getBot().getUsername());
```

Comment:
Let's clean this code a bit (even if you are not author). We can here simply `return event.getInitiator().getUser().getUsername().equals(this.bdkConfig.getBot().getUsername())` as sub-objects cannot be `null`

Reply:
Indeed but it can be null in unit tests.